<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>BmsEditDialog</class>
 <widget class="QDialog" name="BmsEditDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>530</width>
    <height>467</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <item row="0" column="0">
    <widget class="QTabWidget" name="tabWidget_2">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>选择设备</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="1">
        <widget class="QComboBox" name="cboDev">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>80</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>设备</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="2" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="btnSave">
       <property name="text">
        <string>保存</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnOk">
       <property name="text">
        <string>确定</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnCancel">
       <property name="text">
        <string>取消</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="0">
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tabBasicInfo_3">
      <attribute name="title">
       <string>基本信息</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="6" column="5">
        <spacer name="verticalSpacer_9">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="3" column="0" colspan="6">
        <widget class="QGroupBox" name="isCollect">
         <property name="title">
          <string>采集</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>数据来源</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="dataSource">
            <item>
             <property name="text">
              <string>堆</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>组</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>单体</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_33">
            <property name="text">
             <string>变量单位</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="editTagUnit"/>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_36">
            <property name="text">
             <string>功能码</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QComboBox" name="FunctionCodes">
            <item>
             <property name="text">
              <string>0x01</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>0x02</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>0x03</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>0x04</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>0x05</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>0x06</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>0x07</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>0x08</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>0x09</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLabel" name="label_31">
            <property name="text">
             <string>地址</string>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QLineEdit" name="editAddrOffset"/>
          </item>
          <item row="1" column="2">
           <widget class="QLabel" name="label_30">
            <property name="text">
             <string>数值计算</string>
            </property>
           </widget>
          </item>
          <item row="1" column="3">
           <widget class="QLineEdit" name="rate"/>
          </item>
          <item row="2" column="2">
           <widget class="QLabel" name="label_34">
            <property name="text">
             <string>读写类型</string>
            </property>
           </widget>
          </item>
          <item row="2" column="3">
           <widget class="QComboBox" name="cboReadWriteType">
            <item>
             <property name="text">
              <string>只读</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>可写</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_35">
         <property name="text">
          <string>变量名称</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1" colspan="3">
        <widget class="QLineEdit" name="editTagName"/>
       </item>
       <item row="0" column="4">
        <spacer name="horizontalSpacer_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>266</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="2" column="0" colspan="2">
        <widget class="QLabel" name="label_29">
         <property name="text">
          <string>数据类型</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0" colspan="6">
        <layout class="QGridLayout" name="gridLayout_5">
         <item row="0" column="1" colspan="2">
          <widget class="QPlainTextEdit" name="editTagDesc"/>
         </item>
         <item row="0" column="0">
          <layout class="QVBoxLayout" name="verticalLayout_10">
           <item>
            <widget class="QLabel" name="label_37">
             <property name="text">
              <string>变量描述</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer_8">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item row="4" column="0" colspan="6">
        <widget class="QGroupBox" name="isUpload">
         <property name="title">
          <string>发布</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout_6">
          <item row="1" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>计算</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="formula"/>
          </item>
         </layout>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QComboBox" name="cboDataType">
         <item>
          <property name="text">
           <string>U16</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>S16</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>U32</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>S32</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>U64</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>S64</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>FLT</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>DBL</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>VBYTE</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
